third_party/move/move-compiler-v2/legacy-move-compiler/src/parser/syntax.rs

```diff
@@ -1952,7 +1952,24 @@ fn parse_lambda(
         spanned(context.tokens.file_hash(), start_loc, end_loc, vec![])
     };
     let body = Box::new(parse_exp(context)?);
-    Ok(Exp_::Lambda(bindings, body, capture_kind))
+    let spec_opt = if context.tokens.peek() == Tok::Spec {
```

Comment:
Can we please add a test where the spec block is added when the body of the lambda is not enclosed in `{}`, such as: `|x| x spec { ensures result == x }`?

Reply:
done